packages/vite-plugin-cloudflare/src/assets/asset-worker.ts

```diff
@@ -0,0 +1,44 @@
+// @ts-ignore
+import AssetWorker from '@cloudflare/workers-shared/dist/asset-worker.mjs';
+import { UNKNOWN_HOST } from '../shared';
+import type { WorkerEntrypoint } from 'cloudflare:workers';
+
+interface Env {
+	__VITE_ASSET_EXISTS__: Fetcher;
+	__VITE_FETCH_ASSET__: Fetcher;
+}
+
+export default class CustomAssetWorker extends (AssetWorker as typeof WorkerEntrypoint<Env>) {
+	override async fetch(request: Request): Promise<Response> {
+		const response = await super.fetch!(request);
+		const headers = new Headers(response.headers);
+		headers.delete('ETag');
+		headers.delete('Cache-Control');
+
+		// Do we need to do anything else here to duplicate the original response?
+		return new Response(response.body, {
+			status: response.status,
+			headers,
+		});
+	}
+	async unstable_getByETag(
+		eTag: string,
+	): Promise<{ readableStream: ReadableStream; contentType: string }> {
+		const url = new URL(eTag, UNKNOWN_HOST);
+		const response = await this.env.__VITE_FETCH_ASSET__.fetch(url);
+
+		if (!response.body) {
+			throw new Error(`Unexpected error. No HTML found for ${eTag}.`);
```

Comment:
Perhaps return a 500 response rather than throwing?

Reply:
We can't really return a response here because the return type is `Promise<{ readableStream: ReadableStream; contentType: string }>`. The error should only occur if there's a regression in the code so would hopefully be caught in CI rather than by end users.

Reply:
Fair enough, I see that we are throwing in production too: https://github.com/cloudflare/workers-sdk/blob/c1022d66c6536c9e8199ebffe73d2358aa4064ee/packages/workers-shared/asset-worker/src/index.ts#L149-L154